cmd/lakefs/cmd/run.go

```diff
@@ -112,27 +113,60 @@ var runCmd = &cobra.Command{
 
 		// validate service names and turn on the right flags
 		dbParams := cfg.GetDatabaseParams()
-		dbPool := db.BuildDatabaseConnection(ctx, dbParams)
-		defer dbPool.Close()
-		if err := db.ValidateSchemaUpToDate(ctx, dbPool, dbParams); errors.Is(err, db.ErrSchemaNotCompatible) {
-			logger.WithError(err).Fatal("Migration version mismatch, for more information see https://docs.lakefs.io/deploying-aws/upgrade.html")
-		} else if errors.Is(err, migrate.ErrNilVersion) {
-			logger.Debug("No migration, setup required")
-		} else if err != nil {
-			logger.WithError(err).Warn("Failed on schema validation")
-		}
 
-		lockdbPool := db.BuildDatabaseConnection(ctx, dbParams)
-		defer lockdbPool.Close()
+		var (
+			kvStore    kv.Store
+			kvParams   params.KV
+			dbPool     db.Database
+			lockDBPool db.Database
+			err        error
+		)
+		if dbParams.KVEnabled {
+			kvParams = cfg.GetKVParams()
+			kvStore, err = kv.Open(ctx, kvParams)
+			if err != nil {
+				logger.WithError(err).Fatal("Failed to open KV store")
+			}
+			defer kvStore.Close()
 
+			// Check if migration required only on postgres
+			migrationRequired := false
+			if dbParams.Type == kvpg.DriverName {
+				dbPool = db.BuildDatabaseConnection(ctx, dbParams)
+				defer dbPool.Close()
+				_, _, err := db.MigrateVersion(ctx, dbPool, dbParams)
+				if err == nil {
+					migrationRequired = true
+				} else if !errors.Is(err, migrate.ErrNilVersion) {
+					logger.WithError(err).Fatal("Failed to get schema version")
+				}
```

Comment:
What will happen if we start lakefs with kv enabled with blank postgress - it will fail?

Reply:
No - since we will get ErrNilValue - so we won't go into this if clause